great_expectations/core/domain_types.py

```diff
@@ -23,10 +23,12 @@ def has_member_key(cls, key: Any) -> bool:
         return False
 
 
-class StorageDomainTypes(DomainTypes):
+class StructuredDomainTypes(DomainTypes):
```

Comment:
Awesome! Minor nit, but I'm debating whether we want `StructuredDomainTypes` or `StructureDomainType`. I had been thinking the latter, but it might be worth some discussion.

Reply:
@talagluck I prefer "Structured" because it is an adjective -- I think we want that, instead of a noun here.  Thoughts?

Reply:
I understand. When I think of Structured, though, I think that SemanticDomainTypes are also structured. I think the differentiating feature for me is that these DomainTypes refer to structures rather than specifically being structured.